guide/samples/src/pagination.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Examples showing itearting Google API List methods with paginator.
+
+pub async fn paginator_iterate_pages(project_id: &str) -> crate::Result<()> {
+    use google_cloud_gax::paginator::Paginator as _;
+    use google_cloud_secretmanager_v1 as secret_manager;
+
+    let client = secret_manager::client::SecretManagerService::builder()
+        .build()
+        .await?;
+
+    // ANCHOR: paginator-iterate-pages
+    let mut list = client
+        .list_secrets(format!("projects/{project_id}"))
+        .paginator()
+        .await;
+    while let Some(page) = list.next().await {
+        let page = page?;
+        println!("  next_page_token={}", page.next_page_token);
+        page.secrets
+            .into_iter()
+            .for_each(|secret| println!("    secret={}", secret.name));
+    }
+    // ANCHOR_END: paginator-iterate-pages
+
+    Ok(())
+}
+
+pub async fn paginator_stream_pages(project_id: &str) -> crate::Result<()> {
+    use futures::stream::StreamExt;
+    use google_cloud_gax::paginator::Paginator as _;
+    use google_cloud_secretmanager_v1 as secret_manager;
+
+    let client = secret_manager::client::SecretManagerService::builder()
+        .build()
+        .await?;
+
+    // ANCHOR: paginator-stream-pages
+    let list = client
+        .list_secrets(format!("projects/{project_id}"))
+        .paginator()
+        .await
+        .into_stream();
+
+    let _secrets = list
+        .enumerate()
+        .inspect(|(index, page)| {
+            println!(
+                "page={}, next_page_token={}",
+                index,
+                page.as_ref().unwrap().next_page_token
+            )
+        })
+        .collect::<Vec<_>>()
+        .await;
```

Comment:
Seems sad to accumulate the values and then do nothing with them.  Maybe a good use for `fold()`?

```suggestion
    let _ = list
        .enumerate()
        .inspect(|(index, page)| {
            println!(
                "page={}, next_page_token={}",
                index,
                page.as_ref().unwrap().next_page_token
            )
        })
        .fold((), async |u, _unused| u)
        .await;
```

or maybe we can capture the first error instead of calling `unwrap()`?

```rust
    list
        .enumerate()
        .map(|(index, page)| -> Result<()> {
            println!(
                "page={}, next_page_token={}",
                index,
                page?.next_page_token
            );
            Ok(())            
        }
        .fold(Ok(()), async |acc, result| -> Result<()> {
            acc.and(result)
        })
        .await?;
```